provision/minikube/measure/measure-memory-usage.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+set -e
+
+measurementTimes=${1:-"30 60 90 120"}
+podDetectionTimeout=${2:-60}
+measuredPods=""
+export timestamp="${timestamp:-$(date -uIseconds)}"
+
+function measurePodMemory {
+  pod=$1
+  podUID=$(kubectl -n keycloak get pods/${pod} -o jsonpath='{.metadata.uid}')
+  echo "Measuring memory usage for pod '$pod', UID: ${podUID}."
+
+  until ! kubectl -n keycloak get pods/${pod} | grep -i Terminating && [ "$(kubectl -n keycloak get pods/${pod} -o jsonpath='{.status.conditions[?(@.type=="Initialized")].status}' | tr '[:upper:]' '[:lower:]')" == "true" ]; do
+    echo "Waiting for pod '$pod' to get initialized."
+    sleep 1
+  done
+
+  csvFile="data/${timestamp}/${pod}/memory-usage-cgroups.csv"
+  csvHeader="Pod UID"
+  csvData="${podUID}"
+
+  csv2File="data/${timestamp}/${pod}/memory-usage-rss.csv"
+  csv2Header="Pod UID"
+  csv2Data="${podUID}"
+
+  timeInitialized=$(kubectl -n keycloak  get pods/${pod} -o jsonpath='{.status.conditions[?(@.type=="Initialized")].lastTransitionTime}')
+  for measurementTime in $measurementTimes; do
+    csvHeader+=",Memory at ${measurementTime} s (B)"
+    csv2Header+=",RSS at ${measurementTime} s (B)"
+    timeSinceInitialized=$(( $( date +%s ) - $( date -d "$timeInitialized" +%s ) ))
+    podRemainingMeasurementTime=$(( $measurementTime - $timeSinceInitialized ))
+    if [ ${podRemainingMeasurementTime} -lt 0 ]; then
+      podRemainingMeasurementTime=0
+      echo "Pod '${pod}' already exceeded measurement time of ${measurementTime} seconds after start. Skipping."
```

Comment:
looks like when I run it, it spits out the `Pod already exceeded measurement time` message by default

```
]$ ./measure-memory-usage.sh
Opening pod detection window for 60 seconds.
Detected new pod 'keycloak-0'.
Detected new pod 'keycloak-1'.
Measuring memory usage for pod 'keycloak-0', UID: 50715a08-3f34-4ac5-8c1c-0224fa1ca1fe.
Measuring memory usage for pod 'keycloak-1', UID: 812e372d-ab34-4fda-9091-09a9c74b02a0.
Pod 'keycloak-1' already exceeded measurement time of 30 seconds after start. Skipping.
Pod 'keycloak-0' already exceeded measurement time of 30 seconds after start. Skipping.
Pod 'keycloak-1' already exceeded measurement time of 60 seconds after start. Skipping.
Pod 'keycloak-0' already exceeded measurement time of 60 seconds after start. Skipping.
Pod 'keycloak-1' already exceeded measurement time of 90 seconds after start. Skipping.
Pod 'keycloak-0' already exceeded measurement time of 90 seconds after start. Skipping.
Pod 'keycloak-1' already exceeded measurement time of 120 seconds after start. Skipping.
Pod 'keycloak-0' already exceeded measurement time of 120 seconds after start. Skipping.
Measuring memory usage for pod 'keycloak-1' finished.
Measuring memory usage for pod 'keycloak-0' finished.


```